### ISSUE TITLE
nwjs 0.13 beta3 - chrome.fileSystem.restoreEntry breaks retainID on app reload

### ISSUE THREAD

**Issue by cTn-dev** [nwjs/nw.js#4150]
I wrote a "little" demo code to demonstrate this bug, it might be little hard to follow since i mostly ripped out code from my project, but with my notes it should be easy enough to follow.
https://dl.dropboxusercontent.com/u/31740/storage/demo_retain.zip

Simple rundown:
The retainID from chrome.fileSystem.retainEntry api becomes invalid when user uses the "reload app, simulate browser restart" so i guess this issue only affects the SDK build.

About the demo, when you start the app, go to console, call INFORMATION_EXPORTER.selectDirectory()
this will open an UI dialog, choose any folder you like, click ok.
This will create an info file in the directory, just to verify that the retainID is working fine, you can now close the APP, start it again, open console (you will see that the file was initialized again) proving that the file is still accessible.

Now, right click on the app, restart.
Open console and you will be greeted with several errors, but the one you should focus on is
Unchecked runtime.lastError while running fileSystem.restoreEntry: Unknown id

the chrome.storage.local is still working fine, returning the ID (which is just a string) from the local storage, but as something has most likely removed the ID that was returned by chrome.fileSystem.retainEntry the chrome.fileSystem.restoreEntry will fail.

This shouldn't happen.

**Comment by ghostoy** [comment]
Could you post a sample here ?

**Comment by rogerwang** [comment]
This is now fixed in git.
